### ISSUE TITLE
Label: file, "file" or "File"?

### ISSUE THREAD

**Issue by GegznaV** [r-hyperspec/hySpc.read.jdx#24]
In this line:

```r
spc@label$filename <- file
```

Should the label be a file name, word "file" or "File"? I vote for "File".

**Comment by cbeleites** [comment]
Whatever label we choose in the end, I suggest implementing this  in hyperSpec::.spc_io_postprocess_optional.spc_io_postprocess_optional()` so it's consistent across all file import functions. 

- using `"filename"` doesn't need any change as missing labels default to the column name
- I vote for lower case (maybe @bryanhanson has a native speaker opinion on title case vs. lower case)

**Comment by bryanhanson** [comment]
I'm a little confused.  This issue has been raised in `jdx` but @GegznaV it sounds like you are suggesting a universal change in all `hyperSpec` objects.  Is that true?  If true, why change it?  `filename` is clear (specifically, it is clear that is the name of the file and not some how the contents of the file).

Also, `spc_io_postprocess_optional` is not called in `read_jdx` I assume it should be added, but someone please confirm.

**Comment by cbeleites** [comment]
> Also, `spc_io_postprocess_optional` is not called in `read_jdx` I assume it should be added, but someone please confirm.

yes, I added #26

**Comment by cbeleites** [comment]
@bryanhanson I understand @GegznaV to refer to the `@labels` slot which holds labels and units for the various columns.
Not whether the extra data column should be getting a different name from `$filename`

**Comment by GegznaV** [comment]
> I understand @GegznaV to refer to the `@labels` slot 

There are 2 similar lines of code:
```r
    spc@data$filename <- file
    spc@label$filename <- file
```

And yes, I'm referring to just `@labels` and not  `@data`.

E.g., what label should be then, if we bind 2 or more `hyperSpec` objects which have spectra from different files? The values should change and the labels should not.

**Comment by GegznaV** [comment]
> using "filename" doesn't need any change as missing labels default to the column name

So it would be sufficient to remove the lines (there are two of them now):
```r
spc@label$filename <- file
```

That's OK for me.

**Comment by GegznaV** [comment]
Closed via #29
